interface/billing/sl_eob_search.php

```diff
@@ -1,849 +1,832 @@
-      if (! $DEBUG && ! $_POST['form_without']) {
-        sqlStatement("UPDATE form_encounter SET " .
-          "last_stmt_date = '$today', stmt_count = stmt_count + 1 " .
-          "WHERE id = " . $row['id']);
-      }
-      if ($_POST['form_portalnotify']) {
-        if( ! is_auth_portal($stmt['pid']) ){
-            $alertmsg = xlt('Notification FAILED: Not Portal Authorized');
-            break;
+      $pdf2->WriteHTML($content);
+      $temp_filename = $STMT_TEMP_FILE_PDF;
+      $content_pdf = $pdf2->Output($STMT_TEMP_FILE_PDF,'F');
+    } else {
+    $pdf = new Cezpdf('LETTER');//pdf creation starts
+    $pdf->ezSetMargins(45,9,36,10);
+    $pdf->selectFont($GLOBALS['fileroot'] . "/library/fonts/Courier.afm");
+    $pdf->ezSetY($pdf->ez['pageHeight'] - $pdf->ez['topMargin']);
+    $countline=1;
+    $file = fopen($file_to_send, "r");//this file contains the text to be converted to pdf.
+      while(!feof($file)) {
+      $OneLine=fgets($file);//one line is read
+       if(stristr($OneLine, "\014") == true && !feof($file))//form feed means we should start a new page.
+        {
+          $pdf->ezNewPage();
+          $pdf->ezSetY($pdf->ez['pageHeight'] - $pdf->ez['topMargin']);
+          str_replace("\014", "", $OneLine);
         }
-       $inv_count += 1;
-       $pvoice[] = $stmt;
-        // we don't want to send the portal multiple invoices, thus this. Last invoice for pid is summary.
-        if($inv_pid[$inv_count] != $inv_pid[$inv_count+1]){ 
-          fwrite($fhprint, make_statement($stmt));
-            if( !notify_portal($stmt['pid'], $pvoice, $STMT_TEMP_FILE, $stmt['pid'] . "-" . $stmt['encounter'])){
-               $alertmsg = xlt('Notification FAILED');
-                break;
-            }
-            $pvoice = array();
-            flush();
-            ftruncate($fhprint,0);
-       }
-        else    continue;
+      
+      if(stristr($OneLine, 'REMIT TO') == true || stristr($OneLine, 'Visit Date') == true || stristr($OneLine, 'Future Appointments') == true || stristr($OneLine, 'Current') == true )//lines are made bold when 'REMIT TO' or 'Visit Date' is there.
+       $pdf->ezText('<b>'.$OneLine.'</b>', 12, array('justification' => 'left', 'leading' => 6)); 
+      else
+       $pdf->ezText($OneLine, 12, array('justification' => 'left', 'leading' => 6)); 
+       
+      $countline++; 
+     }
+      
+      $fh = @fopen($STMT_TEMP_FILE_PDF, 'w');//stored to a pdf file
+      if ($fh) {
+        fwrite($fh, $pdf->ezOutput());
+        fclose($fh);
+      }
     }
-    
-  } // end while
-
-    if (!empty($stmt)) ++$stmt_count;
-    fwrite($fhprint, make_statement($stmt));
-    fclose($fhprint);
+    header("Pragma: public");//this section outputs the pdf file to browser
+    header("Expires: 0");
+    header("Cache-Control: must-revalidate, post-check=0, pre-check=0");
+    header("Content-Type: application/force-download");
+    header("Content-Length: " . filesize($STMT_TEMP_FILE_PDF));
```

Comment:
`header("Content-Length: ".filesize($file));`
We do this forced download PDF thing in a number of places.  It should be turned into a separate library function.  We also still use at least a couple of different PDF output libraries.  No XPS alternative yet (don't really want it anyway) and PostScript is still just Adobe and is too big.  Wish we had another alternative